activerecord/lib/active_record/associations/preloader/association.rb

```diff
@@ -54,21 +75,15 @@ def owner_keys
           end
 
           def owners_by_key
-            unless defined?(@owners_by_key)
-              @owners_by_key = owners.each_with_object({}) do |owner, h|
-                key = convert_key(owner[owner_key_name])
-                h[key] = owner if key
-              end
+            @owners_by_key ||= owners.each_with_object({}) do |owner, result|
+              key = convert_key(owner[owner_key_name])
+              (result[key] ||= []) << owner if key
             end
-            @owners_by_key
           end
 
           def key_conversion_required?
-            unless defined?(@key_conversion_required)
-              @key_conversion_required = (association_key_type != owner_key_type)
-            end
-
-            @key_conversion_required
+            return @key_conversion_required if defined?(@key_conversion_required)
+            @key_conversion_required = (association_key_type != owner_key_type)
```

Comment:
Is there any reason to touch the code? This is completely the same logic as before.